tlsobs-api/handlers.go

```diff
@@ -430,6 +431,71 @@ func PathsHandler(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+// TruststoreHandler handles the /truststore endpoint of the api.
+// It queries the database for all certificates trusted by a certain program.
+// It takes the following parameters as HTTP query parameters:
+//     store: one of {"mozilla", "android", "apple", "microsoft", "ubuntu"}
+//     format: one of {"json", "pem"}
+func TruststoreHandler(w http.ResponseWriter, r *http.Request) {
+	setResponseHeader(w)
+	log := logger.GetLogger()
+	log.WithFields(logrus.Fields{
+		"form values": r.Form.Encode(),
+		"headers":     r.Header,
+		"uri":         r.URL.RequestURI(),
+	}).Debug("Truststore Endpoint received request")
+
+	val := r.Context().Value(dbKey)
+	if val == nil {
+		httpError(w, http.StatusInternalServerError, "Could not find database handler in request context")
+		return
+	}
+	db := val.(*pg.DB)
+	certs, err := db.GetAllCertsInStore(r.FormValue("store"))
+	if err == pg.ErrInvalidCertStore {
+		httpError(w, http.StatusBadRequest, fmt.Sprintf("Invalid certificate trust store provided: %s", r.FormValue("store")))
+		return
+	} else if err != nil {
+		logrus.Error("Error querying truststore:", err)
+		httpError(w, http.StatusBadRequest, "Error querying trust store")
+		return
+	}
+	switch r.FormValue("format") {
+	case "json":
+		certsJSON, err := json.Marshal(certs)
+		if err != nil {
+			httpError(w, http.StatusInternalServerError, "Could not marshal certificates")
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(certsJSON)
+		logRequest(r, http.StatusOK, len(certsJSON))
+	case "pem":
+		var buffer bytes.Buffer
+		for _, cert := range certs {
+			x509, err := cert.ToX509()
```

Comment:
Rather than re-parsing those values, you can just do the following:

```go
`# Subject: %s
# Issuer: %s`, cert.Subject.String(), cert.Issuer.String()
```